src/main/scala/interfaces/Verification.scala

```diff
@@ -64,22 +69,25 @@ sealed abstract class NonFatalResult extends VerificationResult {
    */
   def &&(other: => VerificationResult): VerificationResult = {
     val r: VerificationResult = other
-    r.combine(this)
+    r.previous = (r.previous :+ this) ++ this.previous
+    r
   }
 }
 
 case class Success() extends NonFatalResult {
+  override var continueVerification: Boolean = true
   override val toString = "Success"
 }
 
 case class Unreachable() extends NonFatalResult {
+  override var continueVerification: Boolean = true
   override val toString = "Unreachable"
 }
 
 case class Failure/*[ST <: Store[ST],
                    H <: Heap[H],
                    S <: State[ST, H, S]]*/
-                  (message: VerificationError)
+                  (message: VerificationError, var continueVerification: Boolean)
```

Comment:
Case classes with `var`s are notoriously error-prone since case classes are by convention immutable. Does the value really need to change?

Second comment: with a default value `continueVerification: Boolean = true`, you would not have to explicitly pass `true` at call site.